src/obj_proc.cpp

```diff
@@ -1909,6 +1913,83 @@ int pull_proc(struct char_data*ch, struct obj_data *obj, int cmd, char*arg, CHAR
 
    return eSUCCESS;
 }
+
+int szrildor_pass(struct char_data *ch, struct obj_data *obj, int cmd, char *arg, CHAR_DATA *invoker)
+{
+        struct obj_data *p;
+        // 30097
+        if (cmd) return eFAILURE;
+
+        if (obj->obj_flags.timer == 1800)
+        {       // Just created - check if this is the first pass in existence and if so, repop zone 161
+                bool first = TRUE;
+                for (p = object_list; p;p = p->next)
+                {
+                        if (obj_index[p->item_number].virt == 30097 && p != obj && p->obj_flags.timer != 1800)  // if any exist that are not at 1800 timer
+                        {
+                                first = FALSE;
+                                break;
+                        }
+                }
+                if (first && real_room(30000) != -1)
+                {
+                        reset_zone(world[real_room(30000)].zone);
+                }
+
+        }
+
+        obj->obj_flags.timer--;
```

Comment:
Doesn't this mean the timer starts as soon as they get the pass, rather than once they use it?   That could be fine, but I thought the quest text indicated it started the timer when they used it, not when they got it. (I only read it once quickly so forgive me if I remembered wrong)

Reply:
I think that's how it's supposed to work :)

Reply:
@dmhenderson76 Can you confirm?